extensions/schema-compare/src/dialogs/schemaCompareDialog.ts

```diff
@@ -46,18 +46,20 @@ export class SchemaCompareDialog {
 	private targetIsDacpac: boolean;
 	private database: string;
 	public dialogName: string;
+	private connection: azdata.connection.ConnectionProfile;
 
 	protected initializeDialog(): void {
 		this.schemaCompareTab = azdata.window.createTab(SchemaCompareLabel);
 		this.initializeSchemaCompareTab();
 		this.dialog.content = [this.schemaCompareTab];
 	}
 
-	public openDialog(p: any, dialogName?: string): void {
+	public async openDialog(p: any, dialogName?: string): Promise<void> {
 		let profile = p ? <azdata.IConnectionProfile>p.connectionProfile : undefined;
 		if (profile) {
 			this.database = profile.databaseName;
 		}
```

Comment:
We're setting the database based on the profile but not the connection? Shouldn't we be using the ID of the profile passed in to set the connection instead?

Reply:
The profile is null if schema compare is launched from the command palette, but there could still be an active connection that should be the default when the dialog is opened.

Reply:
Then you should only get the connection if p is undefined - and should also set the database correctly in that case